src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -304,7 +304,7 @@ private function parseXmlAttributes(\DOMNode $node, array $context = [])
         $typeCastAttributes = $this->resolveXmlTypeCastAttributes($context);
 
         foreach ($node->attributes as $attr) {
-            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes || (isset($attr->nodeValue[1]) && '0' === $attr->nodeValue[0])) {
+            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes || (isset($attr->nodeValue[1]) && '0' === $attr->nodeValue[0] && '.' !== $attr->nodeValue[1])) {
```

Comment:
what's about negative numbers? `-01` ?

Reply:
We still get `-1.0` as float. Exactly how it is now, without this change.

But to be honest I'm not sure what I would expect to receive after decoding it. Considering #32438, probably `"-01"` as a string, but I'm not sure.
Also, one can always use the `xml_type_cast_attributes` context option to force getting a string back.

Reply:
"-01" looks like a string to me.